### PR TITLE
Changes to Roll and Host text and definitions

### PR DESCRIPTION
**Pull request by mcr**

### CODE REVIEW

File: draft-ietf-rats-architecture.md [hunk 1]
```diff
@@ -91,22 +91,16 @@ a remote peer (the "Relying Party") to decide whether to consider that
 Attester a trustworthy peer or not.
 RATS are facilitated by an additional vital party, the Verifier.
 
-The Verifier appraises Evidence via Appraisal Policies and creates
-the Attestation Results to support Relying Parties in their decision
-process.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:96]: Why are these lines deleted?  Recommend keeping them.

File: draft-ietf-rats-architecture.md [hunk 2]
```diff
@@ -91,22 +91,16 @@ a remote peer (the "Relying Party") to decide whether to consider that
...
-purpose is to enable readers of this document to map their current and
-emerging solutions to the architecture provided and the corresponding
-terminology defined.
-
-A common terminology that provides a well-understood semantic meaning to
-the concepts, roles, and models in this document is vital to create
-semantic interoperability between solutions and across different platforms.
+the "Passport Model" and the "Background-Check Model" are illustrated.
+The purpose is to enable readers to map their solution architecture to the
+canonical attestation architecture provided here and to define useful terminology for attestation.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:100]: could be misread as "The purpose is to enable readers to ... define useful terminology for attestation." which is not the intent.

```suggestion
The purpose is to define useful terminology for attestation and enable readers to map
their solution architecture to the canonical attestation architecture provided here.
```

File: draft-ietf-rats-architecture.md [hunk 3]
```diff
@@ -91,22 +91,16 @@ a remote peer (the "Relying Party") to decide whether to consider that
...
+Having a common terminology that provides well-understood meanings for common themes
+such as, roles, device composition, topological models and appraisal is vital for
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:102]:
```suggestion
such as roles, device composition, topological models, and appraisal is vital for
```

File: draft-ietf-rats-architecture.md [hunk 4]
```diff
@@ -130,43 +124,44 @@ Appraisal Policy for Attestation Result:
 
 Attestation Result:
 
-: The output generated by a Verifier, typically including information about an Attester, where the Verifier vouches for the validity of the results
+: The output generated by a Verifier, typically including information about an Attester, where the Verifier vouches for the validity of the Evidence it has appraised
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:127]: I'm not sure the change is technically correct.  The verifier vouches for the level of trustworthiness of the attester, and it vouches for the results it signs.  But it could (for example) ignore extra claims in the evidence (e.g., GPS location) that it cannot vouch for, and provide only vouched-for claims in the results.  Hence it need not vouch for the validity of the entire evidence.   I recommend reverting.

File: draft-ietf-rats-architecture.md [hunk 5]
```diff
@@ -130,43 +124,44 @@ Appraisal Policy for Attestation Result:
...
+: An entity (typically a device), whose Evidence must be appraised in order to infer the extent to which the Attester is considered trustworthy, such as when deciding whether it is authorized to perform some operation
 
 Claim:
 : A piece of asserted information, often in the form of a name/value pair.
 (Compare /claim/ in {{RFC7519}})
 
 Endorsement:
 
-: A secure statement that some entity (typically a manufacturer) vouches for the integrity of an Attester's signing capability
+: Statements that Endorsers make (typically a manufacturer) that vouches for the design and implementation of the Attester. Often this includes statements about the integrity of an Attester's signing capability
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:139]: I think this change is less correct than the original text.  The point about the "signing capability" typically refers to the Attesting Environment, and the endorsement vouches for that environment but not the target environment.   The new proposed text implies that it also vouches for target environments, which I think is not correct.  Recommend reverting.

File: draft-ietf-rats-architecture.md [hunk 6]
```diff
@@ -130,43 +124,44 @@ Appraisal Policy for Attestation Result:
...
 
 Endorser:
 
-: An entity that creates Endorsements that can be used to help to appraise the trustworthiness of Attesters
+: An entity (typically a manufacturer) whose Endorsements help Verifiers appraise the authenticity of Evidence
 
 Evidence:
 
-: A set of information about an Attester that is to be appraised by a Verifier
+: A set of information that asserts the trustworthiness status of an Attester, that is appraised by a Verifier
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:147]: I believe the new text is less correct than the original.   For example, I would not say that a GPS claims is really about the "trustworthiness status" of an Attester, per se.   It's certainly information about an Attester though.  So I think Evidence can contain some info about the trustworthiness status, as well as additional claims that are not.  (E.g., a public key generated by the target environment that it wants to use for communication with a relying party).   As such, I prefer the original.

File: draft-ietf-rats-architecture.md [hunk 7]
```diff
@@ -130,43 +124,44 @@ Appraisal Policy for Attestation Result:
...
 
 Relying Party Owner:
 
-: An entity, such as an administrator, that is authorized to configure Appraisal Policy for Attestation Results in a Relying Party.
+: An entity (typically an administrator), that is authorized to configure Appraisal Policy for Attestation Results in a Relying Party
 
 Verifier:
 
-: An entity that appraises the validity of Evidence about an Attester
+: An entity (typically a service), that appraises the validity of Evidence about an Attester
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:159]: I'm not sure what "a service" means here.   When you have a Verifying Relying Party, is it a service?

File: draft-ietf-rats-architecture.md [hunk 8]
```diff
@@ -643,6 +624,8 @@ plans to support in the TEEP architecture {{-teep-arch}}.
 
 # Roles and Entities
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:626]: Just make sure we do not merge this PR with both versions (but you knew that).   Right now I think the "Henk version" reads better, and is shorter.

File: draft-ietf-rats-architecture.md [hunk 9]
```diff
@@ -666,6 +649,49 @@ network Attester.
 In essence, an entity that combines more than one role also creates and
 consumes the corresponding conceptual messages as defined in this document.
 
+# Role Hosting and Composition
+
+NED VERSION
+
+The RATS architecture includes the definition of Roles (e.g. Attester, Verifier,
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:656]:
```suggestion
The RATS architecture includes the definition of roles (e.g., Attester, Verifier,
```

File: draft-ietf-rats-architecture.md [hunk 10]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+Relying Party, Endorser) and conceptual messages (e.g., Evidence, Attestation Results,
+Endorsements, Appraisal Policies) that captures canonical attestation behaviors,
+that are common to a broad range of attestation-enabled systems. An entity that
+combines multiple Roles produces and consumes the associated Role Messages.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:660]:
```suggestion
combines multiple roles produces and consumes the associated conceptual messages.
```

File: draft-ietf-rats-architecture.md [hunk 11]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+
+The RATS architecture is not prescriptive about deployment configuration options
+of attestation-enabled systems, therefore the various Roles can be hosted on any
+participating entity. This implies, for a given entity, that multiple Roles could
+be co-resident so that the duties of multiple roles could be performed simultaneously.
+Nevertheless, the semantics of which Role Messages are inputs and outputs to a Role
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:666]:
```suggestion
Nevertheless, the semantics of which conceptual messages are inputs and outputs to a role
```

File: draft-ietf-rats-architecture.md [hunk 12]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+entity remains constant. As a result, the entity can participate as a constituent of the RATS architecture
+while flexibly accommodating the needs of various deployment architectures.
+
+Interactions between Roles do not necessarily require use of Internet protocols.
+They could, for example, use inter-process communication, local system buses,
+shared memory, hypervisors, IP-loopback devices or any communication path
+between the various environments that may exist on the entity that combines multiple Roles.
+
+The movement of Role Messages between locally hosted Roles is referred to as
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:675]:
```suggestion
The movement of conceptual messages between locally hosted roles is referred to as
```

File: draft-ietf-rats-architecture.md [hunk 13]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+NED VERSION
+
+The RATS architecture includes the definition of Roles (e.g. Attester, Verifier,
+Relying Party, Endorser) and conceptual messages (e.g., Evidence, Attestation Results,
+Endorsements, Appraisal Policies) that captures canonical attestation behaviors,
+that are common to a broad range of attestation-enabled systems. An entity that
+combines multiple Roles produces and consumes the associated Role Messages.
+
+The RATS architecture is not prescriptive about deployment configuration options
+of attestation-enabled systems, therefore the various Roles can be hosted on any
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:663]:
```suggestion
of attestation-enabled systems, therefore the various roles can be hosted on any
```

File: draft-ietf-rats-architecture.md [hunk 14]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+participating entity. This implies, for a given entity, that multiple Roles could
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:664]:
```suggestion
participating entity. This implies, for a given entity, that multiple roles could
```

File: draft-ietf-rats-architecture.md [hunk 15]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+be co-resident so that the duties of multiple roles could be performed simultaneously.
+Nevertheless, the semantics of which Role Messages are inputs and outputs to a Role
+entity remains constant. As a result, the entity can participate as a constituent of the RATS architecture
+while flexibly accommodating the needs of various deployment architectures.
+
+Interactions between Roles do not necessarily require use of Internet protocols.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:670]:
```suggestion
Interactions between roles do not necessarily require use of Internet protocols.
```

File: draft-ietf-rats-architecture.md [hunk 16]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+They could, for example, use inter-process communication, local system buses,
+shared memory, hypervisors, IP-loopback devices or any communication path
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:672]: re "IP-loopback devices"... if they use "IP" loopback devices, how do they not use "Internet protocols" as in line 670?

File: draft-ietf-rats-architecture.md [hunk 17]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+between the various environments that may exist on the entity that combines multiple Roles.
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:673]:
```suggestion
between the various environments that may exist on the entity that combines multiple roles.
```

File: draft-ietf-rats-architecture.md [hunk 18]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+
+The movement of Role Messages between locally hosted Roles is referred to as
+"local conveyance". Most importantly, the definition of local conveyance methods is
+out-of-scope for the RATS architecture.
+
+The following paragraph elaborates on an exemplary usage scenario:
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:679]: The above sentence is odd, why refer to "the following paragraph" with a header?

File: draft-ietf-rats-architecture.md [hunk 19]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+
+In a Composite Device scenario, in addition to local entities that host the
+lead Attester and other subordinate Attesters, the Composite Device can host the
+Verifier role locally to appraise Evidence from one or more subordinate Attesters.
+The local Verifier might convey local Attestation Results to a remote Relying party
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:684]:
```suggestion
The local Verifier might convey local Attestation Results to a remote Relying Party
```

File: draft-ietf-rats-architecture.md [hunk 20]
```diff
@@ -666,6 +649,49 @@ network Attester.
...
+or the Relying Party role also could become local where an application-specific
+action is taken locally. For example, a secure boot scenario prevents system software
+from loading if the firmware fails to satisfy a local trustworthiness appraisal policy.
+
+In a multi-network scenario, a network node might bridge a wide-area
+network, local-area network, and span various system buses. In so doing, the bridge node
+might need to host multiple Roles depending on the type of behavior each connected domain
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:691]:
```suggestion
might need to host multiple roles depending on the type of behavior each connected domain
```

File: draft-ietf-rats-architecture.md (hunk 1 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:96]: If this was not redundant content, it seems to be vital to highlight that early on.

File: draft-ietf-rats-architecture.md (hunk 2 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:100]: If we are at it, I'd say "remote attestation" instead of "attestation". That nit aside, I think this reads better.

File: draft-ietf-rats-architecture.md (hunk 4 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:127]: I agree with Dave here. And I even think that we should word-smith a proposed change out of this comment and be more verbose here (yes, I know... even more words, but these seem to be universal useful).

File: draft-ietf-rats-architecture.md (hunk 5 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:139]: After reading this more thoroughly again, I have to agree with Dave here, too.

File: draft-ietf-rats-architecture.md (hunk 16 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:672]: True

File: draft-ietf-rats-architecture.md (hunk 18 above)
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:679]: Removing the colon would simplify this into a leading sentence?

File: draft-ietf-rats-architecture.md [hunk 21]
```diff
@@ -91,22 +91,16 @@ a remote peer (the "Relying Party") to decide whether to consider that
 Attester a trustworthy peer or not.
 RATS are facilitated by an additional vital party, the Verifier.
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:93]:
```suggestion
The Verifier appraises Evidence via Appraisal Policies and creates
the Attestation Results to support Relying Parties in their decision
process.
```

File: draft-ietf-rats-architecture.md [hunk 22]
```diff
@@ -90,23 +90,19 @@ produces believable information about itself - Evidence - to enable
 a remote peer (the "Relying Party") to decide whether to consider that
 Attester a trustworthy peer or not.
 RATS are facilitated by an additional vital party, the Verifier.
-
 The Verifier appraises Evidence via Appraisal Policies and creates
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:93]:
```suggestion

The Verifier appraises Evidence via Appraisal Policies and creates
```

File: draft-ietf-rats-architecture.md [hunk 23]
```diff
@@ -130,43 +126,44 @@ Appraisal Policy for Attestation Result:
 
 Attestation Result:
 
-: The output generated by a Verifier, typically including information about an Attester, where the Verifier vouches for the validity of the results
+: The output generated by a Verifier, typically including information about an Attester, where the Verifier vouches for the validity of the Evidence it has appraised
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:129]:
```suggestion
: The output generated by a Verifier, typically including information about an Attester, where the Verifier vouches for the validity of the results
```

File: draft-ietf-rats-architecture.md [hunk 24]
```diff
@@ -130,43 +126,44 @@ Appraisal Policy for Attestation Result:
...
+: An entity (typically a device), whose Evidence must be appraised in order to infer the extent to which the Attester is considered trustworthy, such as when deciding whether it is authorized to perform some operation
 
 Claim:
 : A piece of asserted information, often in the form of a name/value pair.
 (Compare /claim/ in {{RFC7519}})
 
 Endorsement:
 
-: A secure statement that some entity (typically a manufacturer) vouches for the integrity of an Attester's signing capability
+: Statements that Endorsers make (typically a manufacturer) that vouches for the design and implementation of the Attester. Often this includes statements about the integrity of an Attester's signing capability
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:141]:
```suggestion
: A secure statement that some entity (typically a manufacturer) vouches for the integrity of an Attester's signing capability
```

File: draft-ietf-rats-architecture.md [hunk 25]
```diff
@@ -130,43 +126,44 @@ Appraisal Policy for Attestation Result:
...
 
 Endorser:
 
-: An entity that creates Endorsements that can be used to help to appraise the trustworthiness of Attesters
+: An entity (typically a manufacturer) whose Endorsements help Verifiers appraise the authenticity of Evidence
 
 Evidence:
 
-: A set of information about an Attester that is to be appraised by a Verifier
+: A set of information that asserts the trustworthiness status of an Attester, that is appraised by a Verifier
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:149]:
```suggestion
: A set of information about an Attester that is to be appraised by a Verifier
```

File: draft-ietf-rats-architecture.md [hunk 26]
```diff
@@ -130,43 +126,44 @@ Appraisal Policy for Attestation Result:
...
 
 Relying Party Owner:
 
-: An entity, such as an administrator, that is authorized to configure Appraisal Policy for Attestation Results in a Relying Party.
+: An entity (typically an administrator), that is authorized to configure Appraisal Policy for Attestation Results in a Relying Party
 
 Verifier:
 
-: An entity that appraises the validity of Evidence about an Attester
+: An entity (typically a service), that appraises the validity of Evidence about an Attester
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:161]:
```suggestion
: An entity that appraises the validity of Evidence about an Attester
```
